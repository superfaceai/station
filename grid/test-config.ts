import {
  NockConfig,
  SuperfaceTest,
  SuperfaceTestConfig,
} from '@superfaceai/testing';

export const nockConfig: NockConfig = {
  testInstance: expect,
};

/* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
export const buildSuperfaceTest = (global as any).buildSuperfaceTest as (
  options?: SuperfaceTestConfig
) => SuperfaceTest;
