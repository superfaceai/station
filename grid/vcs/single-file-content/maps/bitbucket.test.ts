import { SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

describe(`vcs/single-file-content/bitbucket`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = buildSuperfaceTest({
      profile: 'vcs/single-file-content',
      provider: 'bitbucket',
    });
  });

  describe('SingleFileContent', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          useCase: 'SingleFileContent',
          input: {
            owner: 'jakuvacek',
            repo: 'testrepository',
            path: 'README.md',
            ref: 'master',
          },
        })
      ).resolves.toMatchSnapshot();
    });

    it('should map error', async () => {
      await expect(
        superface.run({
          useCase: 'SingleFileContent',
          input: {
            owner: 'jakuvacek',
            repo: 'testrepository',
            path: 'README.md',
            ref: 'made-up',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
