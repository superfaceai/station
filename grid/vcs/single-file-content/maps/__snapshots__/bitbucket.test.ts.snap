// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/single-file-content/bitbucket SingleFileContent should map error 1`] = `
Err {
  "error": "MappedHTTPError: Expected HTTP error
Properties: {
  \\"message\\": \\"Commit not found\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[4].responseHandlers[0].statements[1]
Original Map Location: Line 20, column 7",
}
`;

exports[`vcs/single-file-content/bitbucket SingleFileContent should perform successfully 1`] = `
Ok {
  "value": Object {
    "content": "# README #

This README would normally document whatever steps are necessary to get your application up and running.

### What is this repository for? ###

* Quick summary
* Version
* [Learn Markdown](https://bitbucket.org/tutorials/markdowndemo)

### How do I get set up? ###

* Summary of set up
* Configuration
* Dependencies
* Database configuration
* How to run tests
* Deployment instructions

### Contribution guidelines ###

* Writing tests
* Code review
* Other guidelines

### Who do I talk to? ###

* Repo owner or admin
* Other community or team contact",
    "encoding": "utf-8",
    "size": 565,
  },
}
`;
